### PR TITLE
JetBrains: Scroll to the first highlighted occurence inside code preview

### LINKED ISSUE #36279: JetBrains: Code preview does not scroll to make the highlight visible
When previewing a larger file where the match is not in the first view LOCs, we want the preview component to scroll to the highlight to behave like the native search component. An example for this is the following query:

```
r:^github\.com/sourcegraph/sourcegraph$ file:README.md Enable web view debugging
```

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #36279
Part-of #36222

This was straight-forward to add. While working on this part I also simplified the validation code to use the least required statements. 

## Test plan

https://user-images.githubusercontent.com/458591/173076137-a6996713-d6f4-43cf-98cd-ab68feae38b3.mov

https://user-images.githubusercontent.com/458591/173080207-e45b2a80-cfce-4acb-9774-cc7fa2c96ba7.mov

## App preview:

- [Web](https://sg-web-ps-jetbrains-editor-scroll.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ecwmrbydob.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/jetbrains/src/main/java/com/sourcegraph/find/PreviewPanel.java [hunk 1]
```diff
@@ -122,10 +121,19 @@ private void openInBrowser(@NotNull PreviewContent previewContent) throws URISyn
         }
     }
 
-    private void addHighlights(Editor editor, @NotNull int[][] absoluteOffsetAndLengths) {
+    private void addAndScrollToHighlights(Editor editor, int[][] absoluteOffsetAndLengths) {
```
Comment (philipp-spiess) [inline client/jetbrains/src/main/java/com/sourcegraph/find/PreviewPanel.java:124]: Regarding the removal of `@NotNull`: JetBrains gave me a warning here that primitives can not be marked as null 🤷

**Comment by philipp-spiess** [comment]
Converted to draft as there is a bug for file results.

**Comment by philipp-spiess** [comment]
> Converted to draft as there is a bug for file results.

Fixed now and added a screencast in the PR description as proof.

File: client/jetbrains/src/main/java/com/sourcegraph/find/PreviewPanel.java (hunk 1 above)
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/find/PreviewPanel.java:124]: Yup, that is correct. Not sure how that got there in the first place.

**Review by vdavid — APPROVED** [review]
Great stuff! Nice and elegant!

I'm still unsure about this invalidate-validate-revalidate (also repaint) thing for AWT components. I tried to read the docs but there were too many unknown concepts around it for me, and it was not worth the time investment so far to understand it.

**Comment by philipp-spiess** [comment]
> I'm still unsure about this invalidate-validate-revalidate (also repaint) thing for AWT components. I tried to read the docs but there were too many unknown concepts around it for me, and it was not worth the time investment so far to understand it.

I don't think this is standard behavior here tbh. IMO an AWT component should use validation _internally_, when it changes some properties and needs to be re-drawn, but we have to call it from the outside in the case of the editor because otherwise it just won't render, lol. 🤷

**Pull request merged**